### PR TITLE
ModelAccess strategy

### PR DESCRIPTION
**Pull request by msladek**
https://synjira.atlassian.net/browse/CELDEV-335

### CODE REVIEW

File: src/main/java/com/celements/model/access/ContextExecutor.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  public ContextExecutor<T, E> inWiki(WikiReference wiki) {
+    wiki = References.cloneRef(wiki, WikiReference.class);
+    return this;
+  }
+
+  public T execute() throws E {
+    WikiReference currWiki = getContext().getWikiRef();
+    try {
+      if (wiki != null) {
```
Comment (fpichler) [inline src/main/java/com/celements/model/access/ContextExecutor.java]: I think checkState(wiki != null) is more appropriate to force a proper usage.

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java [hunk 2]
```diff
@@ -132,138 +101,69 @@ private XWikiDocument getDocumentCloneInternal(DocumentReference docRef) {
...
-    return exists;
+    return exists(docRef, DEFAULT_LANG);
   }
 
   @Override
   public boolean exists(DocumentReference docRef, String lang) {
     boolean exists = false;
     if (docRef != null) {
-      exists = existsFromStore(docRef, lang);
+      lang = normalizeLang(lang);
```
Comment (fpichler) [inline src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:141]: please inline normalizeLang(lang) it is only used once on line 142.

File: src/main/java/com/celements/model/access/DefaultXWikiDocumentCreator.java [hunk 3]
```diff
@@ -20,10 +20,10 @@
   private ModelContext context;
 
   @Override
-  public XWikiDocument createWithoutDefaults(DocumentReference docRef) {
+  public XWikiDocument createWithoutDefaults(DocumentReference docRef, String lang) {
```
Comment (fpichler) [inline src/main/java/com/celements/model/access/DefaultXWikiDocumentCreator.java:23]: please readd old api createWithoutDefaults(DocumentReference) and mark deprecated.

File: src/main/java/com/celements/model/access/DefaultModelAccessFacade.java (hunk 2 above)
Comment (msladek) [reply src/main/java/com/celements/model/access/DefaultModelAccessFacade.java:141]: it is also used on line 214

**Comment by msladek** [comment]
> how can we select the modelAccessStrategy in a other project?

is this a requirement as of now? we could make the component configurable when the demand comes up for that because of a project specific strategy implementation, until then I would stick with YAGNI.

please re-review

**Pull request merged**